src/WrappedMToken.sol

```diff
@@ -121,6 +121,32 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         return _wrap(msg.sender, recipient_, _mBalanceOf(msg.sender));
     }
 
+    /// @inheritdoc IWrappedMToken
+    function wrapWithPermit(
+        address recipient_,
+        uint256 amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external returns (uint240 wrapped_) {
+        IMTokenLike(mToken).permit(msg.sender, address(this), amount_, deadline_, v_, r_, s_);
+
+        return _wrap(msg.sender, recipient_, UIntMath.safe240(amount_));
+    }
```

Comment:
It would be more flexible to pass the address of the signer instead of `msg.sender`, this way a user could provide their signature and the wrapping could be performed by another user.
```suggestion
    function wrapWithPermit(
        address owner_,
        address recipient_,
        uint256 amount_,
        uint256 deadline_,
        uint8 v_,
        bytes32 r_,
        bytes32 s_
    ) external returns (uint240 wrapped_) {
        IMTokenLike(mToken).permit(owner_, address(this), amount_, deadline_, v_, r_, s_);

        return _wrap(owner_, recipient_, UIntMath.safe240(amount_));
    }
```

Reply:
Unfortunately, not because the permit is an ERC20 one, so it does not include the recipient of the wrapped M, and would e ripe for abuse if anyone other than the account themselves called the function. It is the same reason why [others](https://github.com/Uniswap/v2-periphery/blob/master/contracts/UniswapV2Router02.sol#L167) enforcee that the `msg.sender` is the account.

Reply:
Ah yes, you could use the signature and send the Wrapped M to the recipient of your choosing.